apps/cloud/src/app.ts

```diff
@@ -19,11 +23,47 @@ app.use(morgan('dev'));
 app.use(cors());
 // Using a body parser for JSON requests.
 app.use(bodyParser.json());
+//app.use(bodyParser.urlencoded({ extended: false }));
+app.use(cookieParser);
+//app.use(express.static('public'));
 
 app.use('/users', userRouter);
 app.use('/messages', messageRouter);
 
 import taskRoute from "./api/task"
 app.use('/tasks', taskRoute);
 
+
+// Security spike
+import securityInit from "./passportSecurity";
+securityInit(app, userStore);
+app.use('/tasksSecured', passport.authenticate('local', { session: false }), taskRoute);
+
+
+app.use(function(req, res, next) {
+    var err = new Error('Not Found');
+    next(err);
+});
+
+if (app.get('env') === 'development') {
+    app.use(function (err: any, req: any, res: any, next: any) {
+        res.status(err.status || 500);
+        res.render('error', {
+            message: err.message,
+            error: err
+        });
+    });
+}
+
+// production error handler
+// no stacktraces leaked to user
+app.use(function (err: any, req: any, res: any, next: any) {
+    res.status(err.status || 500);
+    res.render('error', {
```

Comment:
The cloud app is a strict JSON api, right? We should be sending error metadata in json instead of trying to render a view